soroban-env-host/src/host.rs

```diff
@@ -358,12 +358,12 @@ impl Host {
     /// underlying [`HostImpl`], returning its constituent components to the
     /// caller as a tuple wrapped in `Ok(...)`. If the provided host reference
     /// is not unique, returns `Err(self)`.
-    pub fn try_finish(self) -> Result<(Storage, Budget, Events), Self> {
+    pub fn try_finish(self) -> Result<(Storage, Budget, Option<Events>), Self> {
```

Comment:
Changing the return value to `Option` because events are lazily externalized, it may fail (e.g. due to out of budget).

Reply:
What about something like `Result<(Storage, Budget, Events), (Self, HostError)>` ? Where we return `Self` _and_ a `HostError` in an `Err` if anything goes wrong, and only return the `Ok` version if _nothing_ goes wrong?

Reply:
This is a great idea. Fixed.